core/src/main/java/com/graphhopper/routing/lm/LandmarkStorage.java

```diff
@@ -497,40 +494,72 @@ int getFromWeight(int landmarkIndex, int node) {
      * @return the weight from the specified node to the landmark (specified *as index*)
      */
     int getToWeight(int landmarkIndex, int node) {
-        int res = (int) landmarkWeightDA.getShort((long) node * LM_ROW_LENGTH + landmarkIndex * 4 + TO_OFFSET)
-                & 0x0000FFFF;
-        assert res >= 0 : "Negative to weight " + res + ", landmark index:" + landmarkIndex + ", node:" + node;
-        if (res == SHORT_INFINITY)
-            return SHORT_MAX;
+        int res = landmarkWeightDA.getInt((long) node * LM_ROW_LENGTH + landmarkIndex * 4);
+
+        //the right bits of "res" store the backward value
+        int from = res & FROM_WEIGHT_INF;
```

Comment:
This statement could be moved after the if to avoid this in case of the specific delta (?)